http_handlers_test.go

```diff
@@ -258,6 +299,13 @@ var registerEndpointTests = map[string]struct {
 		status:          200,
 		json:            tinkerbellDataModel,
 	},
+	"single custom endpoint, non-metadata": {
+		customEndpoints: `{"/id": ".id"}`,
+		url:             "/id",
+		remote:          "192.168.1.5",
```

Comment:
hard coded test IP... does it make sense to have a const here for self documenting code.  What does this remote IP reflect?

Reply:
because hegel looks up the ip from the request's `RemoteAddr` (to then query cacher/tink for the hardware data), this field is to manually set the `request.RemoteAddr` in the tests to make pretend that's where the request came from.

code for reference:
- hegel looking up the ip from the request
https://github.com/tinkerbell/hegel/blob/415e1308915b3fec2f7da7543918b895a9c6452a/http_handlers.go#L247
- setting the remote addr in the tests
https://github.com/tinkerbell/hegel/blob/415e1308915b3fec2f7da7543918b895a9c6452a/http_handlers_test.go#L73

Reply:
https://itnext.io/tips-for-writing-self-documenting-code-e54a15e9de2

GO code can seem esoteric as the origins comes from C/C++, thus the code can be terse (even with the naming conventions).  Unlike other languages (ie. PEP20 focuses on explicit over implicit) and have specifics to address these in the language and linter themselves.

With any good code, bugs, misunderstandings and mistakes can be made with good software practice starting with good self documenting code.

https://itnext.io/tips-for-writing-self-documenting-code-e54a15e9de2

Reflects on key principles for self documenting code:

Being readable vs. clever able.
This reduces mistakes, misunderstandings and bugs introduced as such
"Being a good developer doesn’t mean writing the cleverest code, it means being a good teammate"
Use clear variable names
Use small named functions
No magic numbers... extrapolated also to no magic commands (ie. complex SQL commands).
Remember think of your team and the story you are conveying wrt your intent.

These are all excerpts from the PRI side for Go coding standards we have in place
https://wiki.corp.equinix.com/confluence/display/ES/GO+Coding+Standards

Reply:
Ohh, I see. I didn't know that was the name for this. 

I don't think it makes sense to have a constant for a test case value because these values are chosen arbitrarily. The only significance `192.168.1.5` has is that it is the IP used in that specific mock hardware json. To create a const for it might actually mislead others into thinking it's more significant than it is.  
With the way these tests are set up, the name `remote`  _is_ meant to describe what `192.168.1.5` reflects, but I must admit, `remote` was a poorly chosen name on my part. So what I think I should do is rename `remote` to `userIP` so it's clear that `192.168.1.5` (or any other value assigned to that field) refers to the mock user's IP.

Is that reasonable?

Reply:
Actually, it turns out that the `remote` ip could literally be anything and the mocks would still be returned (the ip isn't actually being matched/checked). So I ended up just creating a constant.